### DISCUSSION TITLE
Android 12 native VPN setting ikev2 unable to connect

### DISCUSSION THREAD

**Discussion by kevin5886** [strongswan/strongswan#2049] [Q&A]
Android 12 native VPN setting ikev2 unable to connect

System (please complete the following information):

OS: Debian12
Kernel version : 6.1.0-15-amd64
strongSwan version(COMPILE): 5.9.0
Tested/confirmed with the latest version: no

Some environmental information:
StrongSwan default configuration file :/usr/local/etc/ipsec.conf

cat /usr/local/etc/ipsec.conf
config setup
    uniqueids = yes
conn %default
    compress = yes
    dpdaction = hold
    dpddelay = 30s
    dpdtimeout = 60s
    inactivity = 600s
    rekey = no
    right = %any
    rightsendcert = never
    rightsourceip = 10.0.0.0/24
    ike = aes128-sha1-modp1024,aes128-sha1-modp1536,aes128-sha1-modp2048,aes128-sha256-ecp256,aes128-sha256-modp1024,aes128-sha256-modp1536,aes128-sha256-modp2048,aes256-aes128-sha256-sha1-modp2048-modp4096-modp1024,aes256-sha1-modp1024,aes256-sha256-modp1024,aes256-sha256-modp1536,aes256-sha256-modp2048,aes256-sha256-modp4096,aes256-sha384-ecp384,aes256-sha384-modp1024,aes256-sha384-modp1536,aes256-sha384-modp2048,aes256-sha384-modp4096,aes256gcm16-aes256gcm12-aes128gcm16-aes128gcm12-sha256-sha1-modp2048-modp4096-modp1024,3des-sha1-modp1024!
    esp = aes128-aes256-sha1-sha256-modp2048-modp4096-modp1024,aes128-sha1,aes128-sha1-modp1024,aes128-sha1-modp1536,aes128-sha1-modp2048,aes128-sha256,aes128-sha256-ecp256,aes128-sha256-modp1024,aes128-sha256-modp1536,aes128-sha256-modp2048,aes128gcm12-aes128gcm16-aes256gcm12-aes256gcm16-modp2048-modp4096-modp1024,aes128gcm16,aes128gcm16-ecp256,aes256-sha1,aes256-sha256,aes256-sha256-modp1024,aes256-sha256-modp1536,aes256-sha256-modp2048,aes256-sha256-modp4096,aes256-sha384,aes256-sha384-ecp384,aes256-sha384-modp1024,aes256-sha384-modp1536,aes256-sha384-modp2048,aes256-sha384-modp4096,aes256gcm16,aes256gcm16-ecp384,3des-sha1!
    fragmentation = yes
    forceencaps = yes
    dpdaction = clear
    dpddelay = 300s
    left = %any
    leftsendcert = always
    leftsubnet = 0.0.0.0/0

conn IKEv2-Android
    keyexchange=ikev2
    leftid=@xx.xx.xx.xxx
    leftcert=xx.xx.xx.crt
    rightid=%any
    rightauth=eap-mschapv2
    eap_identity=%identity
    auto=add

conn IKEv2-Apple
    keyexchange=ikev2
    leftid = *.xx.xx.xx
    leftcert = xx.xx.crt
    rightauth=eap-mschapv2
    rightid=*.xx.xx.xx
    auto=add

IOS certificate uses rsa mode
Android certificate uses ecc mode

StrongSwan default secrets file :/usr/local/etc/ipsec.secrets
StrongSwan compilation parameters :
./configure --enable-eap-identity --enable-eap-md5 \
--enable-eap-mschapv2 --enable-eap-tls --enable-eap-ttls --enable-eap-peap \
--enable-eap-tnc --enable-eap-dynamic --enable-eap-radius --enable-xauth-eap \
--enable-xauth-pam --enable-dhcp --enable-openssl --enable-addrblock \
--enable-unity \
--enable-md4 \
--enable-certexpire --enable-radattr --disable-gmp

Describe the question:
As of the current time, I can connect using both iOS and Windows machines and the Android strongswan client.However, using the native VPN method that comes with Android 12 is not available.Can you provide some reference configurations that can meet the common requirements of iOS, Windows, Android native, and Android strongswan clients?

Thank you.I'm looking forward to your reply

**Comment by exoticprgmr** [comment]
From my own experience... Native VPN client on Android is broken (especially on samsung's) - StrongSwan android app works best for me.

If you want to try your luck on native vpn, connect your device via USB (with USB debugging enabled) and call below command to collect logs from your Android client.

`adb logcat -s charon:*`

**Reply by kevin5886** [reply]
Thank you very much for your reply.
We still want to use the native version.This way, we don't have to install app software.It seems more convenient.It should be supported.

**Reply by lucarez** [reply]
@kevin5886, did you have any luck on successfully configuring the vpn? I'm trying to do the same...

**Reply by kevin5886** [reply]
Still failed. I changed the software later on

**Reply by exoticprgmr** [reply]
hope you're using StrongSwan for android. Works perfect for me all time

**Reply by lucarez** [reply]
Actually I'm trying to make it work without installing custom software on the device (either android nor ios)
